index.html

```diff
@@ -2622,178 +2649,285 @@ <h3><dfn>Writing or pushing content</dfn></h3>
+      <p>
+        The [[NFC-RTD]] specifies that external types MUST contain
+        the domain name of the issuing organization, a a colon and a type
+        name that is at least one character long.
+      </p>
+      <p>
+        The [[NFC-RTD]] specifies to omit the URN prefix “`urn:nfc:ext:`”
+        in the binary encoding of external type records, therefore
+        applications SHOULD NOT specify that when creating
+        <a>external type records</a>.
+      </p>
+      <p>
+        When the domain name is not provided, by default Web NFC uses the
+        <a data-cite="html">effective domain</a> of the {{Document/origin}} as
+        domain name in external type names when
+        <a href="#mapping-external-data-to-ndef">
+        creating external type records</a>. Therefore, applications MAY specify
+        only the colon and the external type name.
+      </p>
+      <p class="note">
+        The [[NFC-RTD]] requires that external type names are represented
+        with the URN prefix “`urn:nfc:ext:`”, e.g. when reading
+        <a>NDEF messages</a>. However, since <a>external type records</a> are
+        distinguished by having the <a>TNF FIELD</a> set to `0x04`, there is
+        no risk seen for type name clashing. Also, there are
+        <a href="https://www.w3.org/2001/tag/doc/URNsAndRegistries-50.html">
+        W3C TAG recommendations</a> to avoid using URNs in the Web.
+        Therefore, Web NFC does not use the URN prefix neither when reading
+        or writing <a>NDEF messages</a>, but requires the leading colon ('`:`')
+        before the type name.
+      </p>
+      <p>
+        To <dfn>validate external type</dfn> given a |input:USVString| run
+        these steps:
+      </p>
+      <ol class=algorithm id="validate-external-type">
+        <li>
+          If |input:string| is not a {{USVString}} or it is empty, or its
+          length exceeds 255 bytes, or it contains white spaces,
+          return `false`.
+        </li>
+        <li>
+          If |input| does not start with colon ('`:`'), return |input|.
+          <p class="issue">
+            Validating domain name needs an algorithm.
+          </p>
+        </li>
+        <li>
+          If |input| starts with colon ('`:`'), let |domain:string| be the
+          <a data-cite="url#concept-url-serializer">serialization</a> of the
+          <a data-cite="html">effective domain</a> of the {{Document/origin}}.
+        </li>
+        <li>
+          Return |domain| concatenated with |input|.
+        </li>
+      </ol>
+    </section>
 
-      <section><h3>Mapping empty record to NDEF</h3>
+    <section><h3>Validating local type</h3>
+      <p>
+        To <dfn>validate local type</dfn> given a |input:USVString| run
+        these steps:
+      </p>
+      <ol class=algorithm id="validate-local-type">
+        <li>
+          If |input| is not a {{USVString}} or its length exceeds 255 bytes,
+          return `false` and terminate these steps.
```

Comment:
Please use "abort" instead of "terminate".

Reply:
Noted for the future as well. Is there a difference in terminate vs abort? Abort (for me) means we do clean-up steps as well, possibly rolling back a transaction. Terminate means jumping to the end.

Reply:
Terminology is that specified that precisely to my knowledge, but abort is what most specs use today

Reply:
I'm not sure if there's a difference to be honest.
FYI https://github.com/search?q=org%3AWICG+%22terminate+these+steps%22&type=Code vs https://github.com/search?q=org%3AWICG+%22abort+these+steps%22&type=Code

Reply:
We used both in manifest but mostly moved to abort now

Reply:
OK, so it's internal terminology (not aligned with terms outside web specs).